c10/core/DispatchKey.h

```diff
@@ -84,17 +84,17 @@ enum class DispatchKey : uint8_t {
   // based on the layout of the tensor.  Note that the sparse backends
   // are one case where ordering matters: sparse multi-dispatches with
   // the corresponding dense tensors, and must be handled before them.
-  MkldnnCPUTensorId,  // registered at build/aten/src/ATen/MkldnnCPUType.cpp
-                      // NB: not to be confused with MKLDNNTensorId, which is Caffe2 only
-  SparseCPUTensorId,  // registered at build/aten/src/ATen/SparseCPUType.cpp
-  SparseCUDATensorId, // registered at build/aten/src/ATen/SparseCUDAType.cpp
-  SparseHIPTensorId,  // TODO: I think this is not actually used, due to Note [Masquerading as CUDA]
+  MkldnnCPU,  // registered at build/aten/src/ATen/MkldnnCPUType.cpp
+                      // NB: not to be confused with MKLDNN, which is Caffe2 only
+  SparseCPU,  // registered at build/aten/src/ATen/SparseCPUType.cpp
+  SparseCUDA, // registered at build/aten/src/ATen/SparseCUDAType.cpp
+  SparseHIP,  // TODO: I think this is not actually used, due to Note [Masquerading as CUDA]
 
-  // Here are reserved backends for user-defined backends, see Note [Private use TensorId]
+  // Here are reserved backends for user-defined backends, see Note [Private use DispatchKey]
   // To see some example about how to use this, check out MSNPU
-  PrivateUse1_TensorId,
-  PrivateUse2_TensorId,
-  PrivateUse3_TensorId,
+  PrivateUse1,
```

Comment:
side note: should we call those UserDefined instead of PrivateUse? As a library user, PrivateUse would make me think the library reserved those for some future purpose and I'm not supposed to use them.

Reply:
private use is consistent with nomenclature from, e.g., networking rfcs https://tools.ietf.org/html/rfc1918 The key point is that you shouldn't publish a library that uses these; they are reserved for private use by users